climada/hazard/tc_tracks.py

```diff
@@ -1275,25 +1279,22 @@ def write_hdf5(self, file_name, complevel=5):
             Specifies a compression level (0-9) for the zlib compression of the data.
             A value of 0 or None disables compression. Default: 5
         """
-        # initialise the data set with the size information
-        xr.Dataset(attrs={'size': self.size}).to_netcdf(file_name, mode="w", format="NetCDF4")
-
-        LOGGER.info('Writing %d tracks to %s', self.size, file_name)
-        last_perc = 0
         for i, track in enumerate(self.data):
-            perc = 100 * i / self.size
-            if perc - last_perc >= 10:
-                LOGGER.info("Progress: %d%%", perc)
-                last_perc = perc
-
             # change dtype from bool to int to be NetCDF4-compliant
             track.attrs['orig_event_flag'] = int(track.attrs['orig_event_flag'])
-            encoding = {var: dict(zlib=True, complevel=complevel) for var in track.data_vars}
-            track.to_netcdf(file_name, mode="a", group=f"track{i}", encoding=encoding)
-            track.attrs['orig_event_flag'] = bool(track.attrs['orig_event_flag'])
-
-        if last_perc != 100:
-            LOGGER.info("Progress: 100%")
+        try:
```

Comment:
When using a try block, you probably should do something if there is a particular exception right? At least give a message that it failed or something.

Reply:
No, when using io operations with a handler, it's best practice to have a `try ... finally` clause even if you don't know which kind of exception might happen. This applies to all programming languages that have this clause. In Python, this is usually obfuscated by the `with ...:` clause. There is even a built-in method for the special case of calling the `close` method of an object: https://docs.python.org/3/library/contextlib.html#contextlib.closing

I changed the code to use the contextlib functionality in 2facdaab

Reply:
Makes sense. I did not understand that the goal was to have a `close` method. I like the change to the context manager.